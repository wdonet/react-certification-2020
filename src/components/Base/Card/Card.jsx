import React from 'react';
import styled from 'styled-components';

const StyledDiv = styled.div`
  overflow: hidden;
  height: 345px;
  width: 345px;
  margin: 8px;
  box-shadow: 2px 2px 2px 2px #ccc;
  border-radius: 5px;

  &:hover {
    background: lightgray;
  }
`;

const StyledCardContent = styled.div`
  text-align: justify;
  padding: 4px;
  margin: 4px;
`;

const StyledCardImage = styled.img`
  height: 140px;
  width: 100%;
  display: block;
  background-size: cover;
  background-repeat: no-repeat;
  background-position: center;
  object-fit: cover;
`;

const StyledCardTitle = styled.h2`
  font-weight: normal;
  font-size: 1.25rem;
`;

const StyledCardDescription = styled.p`
  font-weight: lighter;
  color: gray;
`;

const Card = ({ title, image, description }) => {
  return (
    <StyledDiv>
      <StyledCardImage src={image} alt={title} />
      <StyledCardContent>
        <StyledCardTitle>{title}</StyledCardTitle>
        <StyledCardDescription>{description}</StyledCardDescription>
      </StyledCardContent>
    </StyledDiv>
  );
};

export default Card;
