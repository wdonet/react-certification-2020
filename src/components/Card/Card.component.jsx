import React from 'react';
import Styled from './Card.styles';

const Card = ({ title, thumbnail, description }) => (
  <Styled.Container>
    <Styled.TitleBar>
      <Styled.Title>{title}</Styled.Title>
    </Styled.TitleBar>
    <Styled.Content>
      <Styled.Thumbnail src={thumbnail} />
      <Styled.Description>{description}</Styled.Description>
    </Styled.Content>
  </Styled.Container>
);

export default Card;
