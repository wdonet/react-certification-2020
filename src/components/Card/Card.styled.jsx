import styled from 'styled-components';

const CardWrapper = styled.div`
  border-radius: 4px;
  box-shadow: 0px 2px 1px -1px rgb(0 0 0 / 20%), 0px 1px 1px 0px rgb(0 0 0 / 14%), 0px 1px 3px 0px rgb(0 0 0 / 12%);
  height: 345px;
  margin: 10px;
  overflow: hidden;
  width: 345px;
`;

const CardButton = styled.button`
  background: none;
  color: inherit;
  border: none;
  padding: 0;
  font: inherit;
  outline: inherit;
`;

const CardImage = styled.img`
  width: 100%;
  height: 140px;
  object-fit: cover;
`;

const CardBody = styled.div`
  text-align: start;
  padding: 16px;
`;

const CardTitle = styled.div`
  font-size: 20px;
  font-family: 'Roboto', 'Helvetica', 'Arial', sans-serif;
  font-weight: 500;
  line-height: 1.6;
  letter-spacing: 0.0075em;
`;

const CardDescription = styled.p`
  font-size: 14px;
  font-family: 'Roboto', 'Helvetica', 'Arial', sans-serif;
  font-weight: 400;
  line-height: 1.43;
  letter-spacing: 0.01071em;
  color: rgba(0, 0, 0, 0.54);
`;

export { CardWrapper, CardButton, CardImage, CardBody, CardTitle, CardDescription };
