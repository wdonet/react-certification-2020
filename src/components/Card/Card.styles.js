import styled from 'styled-components';

const Container = styled.div`
  border: 1px solid #cccccc;
  border-radius: 5px;
  padding: 20px;
  background-color: rgba(230, 90, 232, 0.2);
  min-width: 350px;
  width: 25%;
  display: inline-block;
  height: 500px;

  &:hover {
    cursor: pointer;
  }

  &:not(:last-child) {
    margin-bottom: 20px;
  }
`;
const TitleBar = styled.div`
  height: 85px;
`;

const Title = styled.h2`
  margin-top: 5px;
  font-size: 1em;

  ${Container}:hover & {
    font-size: 1.2em;
  }
`;

const Content = styled.div``;

const Thumbnail = styled.img`
  width: 70%;
  height: auto;
  margin-left: auto;
  margin-right: auto;
  border-radius: 8px;
`;

const Description = styled.p`
  opacity: 0.8;

  &:first-child {
    font-style: italic;
  }
`;

const Styled = { Container, Title, TitleBar, Content, Thumbnail, Description };
export default Styled;
