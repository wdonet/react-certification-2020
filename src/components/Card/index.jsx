import React from "react";
import Styled from "./styled";

function Card(props) {
    return (
        <Styled.Card>
            <Styled.CardImg imgUrl={props.imgUrl} ></Styled.CardImg>
            <Styled.CardContent>
                <Styled.CardTitle> {props.title} </Styled.CardTitle>
                <Styled.CardDescription> {props.description} </Styled.CardDescription>
            </Styled.CardContent>
        </Styled.Card>
    );
  };
  
export default Card;