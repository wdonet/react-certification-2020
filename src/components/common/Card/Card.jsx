// React
import React, { useCallback } from 'react';
// Router
import { useHistory } from 'react-router';
// PropTypes
import PropTypes from 'prop-types';
// Styles
import {
  CardContainer,
  CardImage,
  CardTitle,
  CardInnerContainer,
  CardDescription,
} from './styles';

const Card = ({ image, title, description, url, vertical }) => {
  const history = useHistory();
  const ellipsis = (text, end) => {
    return text && text.length > end ? `${text.slice(0, end)}...` : text;
  };
  const handleClick = useCallback(() => {
    history.push(url);
    // window.open(url, '_self');
  }, [url, history]);

  return (
    <CardContainer data-testid="card-test" onClick={handleClick} vertical={vertical}>
      <CardImage image={image} vertical={vertical} />
      <CardInnerContainer vertical={vertical}>
        <CardTitle vertical={vertical}>{ellipsis(title, 50)}</CardTitle>
        <CardDescription vertical={vertical}>
          {ellipsis(description, 140)}
        </CardDescription>
      </CardInnerContainer>
    </CardContainer>
  );
};

Card.defaultProps = {
  vertical: false,
};

Card.propTypes = {
  image: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
  vertical: PropTypes.bool,
};

export default Card;
