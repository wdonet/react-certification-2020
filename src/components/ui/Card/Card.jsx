import React, { useContext } from 'react';
import styled, { ThemeContext } from 'styled-components';

const StyledDiv = styled.div`
  overflow: hidden;
  height: 345px;
  width: 345px;
  margin: 8px;
  box-shadow: 2px 2px 2px 2px #ccc;
  border-radius: 5px;
  background: ${({ theme }) => theme.color.surface};

  &:hover {
    background: lightgray;
    cursor: pointer;
  }
`;

const Card = (props) => {
  const { onClick } = props;
  const { theme } = useContext(ThemeContext);
  return (
    <StyledDiv onClick={onClick} role={props.role || ''} theme={theme}>
      {props.children}
    </StyledDiv>
  );
};

export default Card;
