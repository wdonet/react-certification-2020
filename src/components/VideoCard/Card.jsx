import React from 'react';
import Card from 'react-bootstrap/Card';

function VideoCard({ src, title, description }) {
  return (
    <Card
      style={{ width: '18rem', height: '30rem', marginBottom: '1rem', marginTop: '1rem' }}
    >
      <Card.Body>
        <Card.Img variant="top" src={src} />
        <Card.Title>{title}</Card.Title>
        <Card.Text>{description}</Card.Text>
      </Card.Body>
    </Card>
  );
}
export default VideoCard;
