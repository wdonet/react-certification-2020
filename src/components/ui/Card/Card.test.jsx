import React from 'react';
import 'jest-styled-components';
import { getByRole } from '@testing-library/react';
import { renderWithTheme } from '../../../utils';
import { lightTheme } from '../../../providers/themes';
import Card from './Card';

const build = (Component = <Card />, theme = lightTheme) => {
  const { container } = renderWithTheme(Component, theme);
  return { container };
};

describe('Card', () => {
  it('renders', () => {
    const container = build();
    expect(container).toMatchSnapshot();
  });
});

describe('Card styles and props', () => {
  it('applies default styling', () => {
    const { firstChild } = build().container;
    expect(firstChild).toHaveStyle('overflow: hidden');
    expect(firstChild).toHaveStyle('height: 345px');
    expect(firstChild).toHaveStyle('width: 345px');
    expect(firstChild).toHaveStyle('margin: 8px');
    expect(firstChild).toHaveStyle('box-shadow: 2px 2px 2px 2px #ccc');
    expect(firstChild).toHaveStyle('border-radius: 5px');
  });

  it('has role and all passed props', () => {
    const EXPECTED_ROLE = 'figure';
    const EXPECTED_CONTENT = "Hi, I'm your card content";
    const { container } = build(<Card role={EXPECTED_ROLE}>{EXPECTED_CONTENT}</Card>);
    expect(getByRole(container, EXPECTED_ROLE)).toBeInTheDocument();
    expect(container).toHaveTextContent(EXPECTED_CONTENT);
  });
});
