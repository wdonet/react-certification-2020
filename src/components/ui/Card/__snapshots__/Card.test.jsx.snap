// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders 1`] = `
Object {
  "container": <div>
    <div
      class="sc-bdVaJa ikyuFl"
      role=""
    />
  </div>,
}
`;
